docs/07-lifecycle.md

```diff
@@ -2,8 +2,54 @@
 ## Development
 ## Submit to Marketplace
 ## Provision
+### App Bundle
+App bundle contains `manifest.json` file, `static/` folder and one or several lambda function bundles.
+- `static/` folder may contain static files app needs. They are automatically provisioned and stored in the AWS S3 bucket. 
+- `manifest.json` file contains details about the app such as appID, appVersion, appType(http or an AWS app), requested permissions, requested locations, and information about the functions.
+- Each of the lambda function bundles is a valid and runnable AWS lambda function, which are provisioned in the AWS by the Gitlab pipeline. 
+
+### Provisioning In The 3rd Party AWS
+Provisioning in the third party AWS cloud environment is done by the **appsctl** which reads appropriate AWS credentials from environment variables:
```

Comment:
we should prob. detail which appsctl command does the actual provisioning  and call out appsctl as a tool/script, etc.